### PR TITLE
refactor rule metadata in java backend to use typed attributes

### PR DESCRIPTION
**Pull request by dwightguth**
@yilongli please review

### CODE REVIEW

**Comment by yilongli** [comment]
lgtm.

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Please re-review. @yilongli @radumereuta

**Comment by yilongli** [comment]
lgtm

File: src/javasources/KTool/src/org/kframework/parser/concrete/disambiguate/CorrectCastPriorityFilter.java [hunk 1]
```diff
@@ -48,6 +48,7 @@ public ASTNode visit(Cast cst, Void _) throws ParseFailedException {
                 var.setUserTyped(true);
                 var.setSort(cst.getSort());
                 var.setSyntactic(cst.getType() != CastType.SEMANTIC);
+                var.copyAttributesFrom(cst);
```
Comment (radumereuta) [inline src/javasources/KTool/src/org/kframework/parser/concrete/disambiguate/CorrectCastPriorityFilter.java:51]: You should do this in `RemoveSyntacticCasts` too.

Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/parser/concrete/disambiguate/CorrectCastPriorityFilter.java:51]: It's done there already

**Comment by radumereuta** [comment]
Ok, it seems that my issues are fixed. Thanks.
lgtm

**Pull request merged**